### PR TITLE
Add ability to the place page api to be able to request data for a place and a category. 

### PR DESCRIPTION
**Pull request by pulkit-s**
Add ability to the place page api to be able to request data for a place and a category. (currently it can only request data for a page).

Corresponding Bug: b/206032102

### CODE REVIEW

**Comment by shifucun** [comment]
Please add golden test case when all the involved cache are rebuild

**Comment by pulkit-s** [comment]
> 

Done

**Review by shifucun — APPROVED** [review]
Thanks for making the change!

**Review by shifucun — COMMENTED** [review]
I can not see the diff in [internal/server/v1/variables/golden/variables/santa_clara.json](https://github.com/datacommonsorg/mixer/pull/896/files#diff-ae0bb09067ba701e9dc5835352f654bd8b7cbe09baf3a83dc95cf1fbe8debcea), can you paste a screenshot on what the addition is

**Comment by pulkit-s** [comment]
image

**Comment by pulkit-s** [comment]
image

**Comment by pulkit-s** [comment]
> I can not see the diff in [internal/server/v1/variables/golden/variables/santa_clara.json](https://github.com/datacommonsorg/mixer/pull/896/files#diff-ae0bb09067ba701e9dc5835352f654bd8b7cbe09baf3a83dc95cf1fbe8debcea), can you paste a screenshot on what the addition is



Added screenshots. Do these look fine?

**Comment by shifucun** [comment]
> > I can not see the diff in [internal/server/v1/variables/golden/variables/santa_clara.json](https://github.com/datacommonsorg/mixer/pull/896/files#diff-ae0bb09067ba701e9dc5835352f654bd8b7cbe09baf3a83dc95cf1fbe8debcea), can you paste a screenshot on what the addition is
> 
> Added screenshots. Do these look fine?

I see, these should be the imputed IPCC stat vars. @pradh @n-h-diaz There are a lot of these IPCC stat var, and maybe data, do we have any concern on the size?

File: internal/store/bigtable/key.go [hunk 1]
```diff
@@ -46,6 +46,8 @@ const (
 	BtPlacesMetadataPrefix = "d/d/"
 	// BtObsCollection for obs collection cache.
 	BtObsCollection = "d/e/"
+  // BtPlacePageCategoricalPrefix for place page Categorical cache.
```
Comment (pradh) [inline internal/store/bigtable/key.go:49]: nit: is the indent right?

File: internal/server/placepage/place_page.go [hunk 2]
```diff
@@ -215,13 +215,21 @@ func fetchBtData(
...
 	statVars []string,
+	category string,
 ) (map[string]*pb.StatVarSeries, map[string]*pb.PointStat, error) {
 	// Fetch place page cache data in parallel.
+	action := [][]string{places}
+	prefix := bigtable.BtPlacePagePrefix
+	if category != "" {
+		action = [][]string{places, {category}}
+		prefix = bigtable.BtPlacePageCategoricalPrefix
+	}
```
Comment (pradh) [inline internal/server/placepage/place_page.go:226]: Generally, for PRs where we're in the process of migrating caches/data, its good to point out in the Description as to why the change is safe.

The change is a NOP until website starts setting `category`.  And before we do that, we should ensure all import-groups are rebuilt with the new sharded cache.

**Comment by pradh** [comment]
> > > I can not see the diff in [internal/server/v1/variables/golden/variables/santa_clara.json](https://github.com/datacommonsorg/mixer/pull/896/files#diff-ae0bb09067ba701e9dc5835352f654bd8b7cbe09baf3a83dc95cf1fbe8debcea), can you paste a screenshot on what the addition is
> > 
> > 
> > Added screenshots. Do these look fine?
> 
> I see, these should be the imputed IPCC stat vars. @pradh @n-h-diaz There are a lot of these IPCC stat var, and maybe data, do we have any concern on the size?

I think its only about 270-odd IPCC SVs were added (cl/469999871).  Because the json file, as is, is very big the diffs are not seen.
